### PR TITLE
Add vetos

### PR DESCRIPTION
**Pull request by Rainyan**
TLDR: ~~This PR isn't really ready to be merged as-is~~ (actual spec-client veto visuals are missing), but could work a starting point for implementing vetos to the overlay.

This PR adds support for accessing the vetos information from spec-client, by using the SM global forwards provided by the [nt_competitive_vetos](https://github.com/Rainyan/sourcemod-nt-competitive-vetos) plugin.

This change adds [these two includes](https://github.com/Rainyan/sourcemod-nt-competitive-vetos/tree/main/scripting/include) as compile dependencies for the nt_websocket plugin. However, nt_websocket binary will continue to work even without the veto plugin present, as it's just listening for the global forwards regardless of if they actually exist or not.

### What works:
* nt_websocket listens for the global forwards _OnMapVetoStageUpdate(...)_ and _OnMapVetoPick(...)_, and sends that data through the websocket using the unoccupied 'Y' and 'Z' control chars.
* spec-client receives that data in _async function handleMessage(...)_, and console.log's it, with output something like:
```
Y -- Stage: CoinFlip // Coin flip begins
Y -- Stage: CoinFlipResult && Team: NSF // NSF wins coin flip
Z -- Stage: FirstTeamBan && Team: NSF && Map: nt_ghost_ctg // NSF bans ghost
Y -- Stage: FirstTeamBan
Z -- Stage: SecondTeamBan && Team: Jinrai && Map: nt_sentinel_ctg // Jinrai bans sentinel
Y -- Stage: SecondTeamBan
Z -- Stage: SecondTeamPick && Team: Jinrai && Map: nt_dusk_ctg // Jinrai picks dusk
Y -- Stage: SecondTeamPick
Z -- Stage: FirstTeamPick && Team: NSF && Map: nt_threadplate_ctg // NSF picks threadplate
Y -- Stage: FirstTeamPick
Z -- Stage: RandomThirdMap && Team: Spectator && Map: nt_skyline_ctg // Random map is skyline
Y -- Stage: RandomThirdMap
Y -- Stage: Inactive // Veto is over
```

### What doesn't work:
* There's no visual indication whatsoever in the spec-client for these vetos. All it does it console.log().

### Setting up fake the players & veto activity for testing:
* Set the `NT_RELAY_DEBUG` define in nt_websocket.sp to true. This compiles in the "sm_relaydbg_populate" debug command, which adds 10 fake clients to the spec-client view, so one can approximate how it would look on an actual server.
* Un-comment the `#define DEBUG_FAKE_VETOS` define in nt_competitive_vetos.sp. This compiles in the "sm_debug_fake_veto" debug command, which will play back a random fake veto process, resulting in spec-client input similar to what was described in those console.log lines above.

I'm not too familiar with Vue.js, so maybe I should leave that up to you to decide how to visually represent? Could probably mimic the visual imagery of vetos in other comp games/streams.

### CODE REVIEW

**Comment by Rainyan** [comment]
Added websocket control char 'L' for sending/receiving the veto map list. The list is sent when the VETO_STAGE_COIN_FLIP stage triggers at the beginning of a brand new veto.

The full spec-client veto input data console.log print now looks something like:
```
L -- Maps:
    array of: [nt_ballistrade_ctg,nt_bullet_tdm,nt_dawn_ctg,
    nt_decom_ctg,nt_disengage_ctg,nt_dusk_ctg,
    nt_engage_ctg,nt_ghost_ctg,nt_isolation_ctg]

Y -- Stage: CoinFlip
Y -- Stage: CoinFlipResult && Team: Jinrai
Z -- Stage: FirstTeamBan && Team: Jinrai && Map: nt_dusk_ctg
Y -- Stage: FirstTeamBan
Z -- Stage: SecondTeamBan && Team: NSF && Map: nt_dawn_ctg
Y -- Stage: SecondTeamBan
Z -- Stage: SecondTeamPick && Team: NSF && Map: nt_decom_ctg
Y -- Stage: SecondTeamPick
Z -- Stage: FirstTeamPick && Team: Jinrai && Map: nt_isolation_ctg
Y -- Stage: FirstTeamPick
Z -- Stage: RandomThirdMap && Team: Spectator && Map: nt_ghost_ctg
Y -- Stage: RandomThirdMap
Y -- Stage: Inactive
```

**Comment by Rainyan** [comment]
Ok, I'm probably done messing with this PR now. This stuff should be good to merge.
